eZ/Publish/Core/Persistence/Legacy/Content/TreeHandler.php

```diff
@@ -248,9 +249,18 @@ public function changeMainLocation($contentId, $locationId)
         );
 
         // Update subtree section to the one of the new main location parent location content
-        $this->setSectionForSubtree(
-            $locationId,
-            $this->loadContentInfo($this->loadLocation($parentLocationId)->contentId)->sectionId
-        );
+        $destinationContentId = $this->loadLocation($parentLocationId)->contentId;
+        try {
+            $sectionId = $this->loadContentInfo($destinationContentId)->sectionId;
```

Comment:
can this operation return null as well? The code would suggest so, but I'm not sure if that's the case.

Reply:
@alongosz I don't think that's the case because `section_id` column is not nullable in the case of `ezcontentobject` table, by loading ContentInfo we load the data from this table so each and every ContentInfo object will have `sectionId` assigned.

Reply:
> @alongosz I don't think that's the case because `section_id` column is not nullable in the case of `ezcontentobject` table, by loading ContentInfo we load the data from this table so each and every ContentInfo object will have `sectionId` assigned.

My thoughts exactly. So instead of `try-catch; if;`, you could simply do:
```php
        try {
            $sectionId = $this->loadContentInfo($destinationContentId)->sectionId;
            $this->setSectionForSubtree(
                $locationId,
                $sectionId
            );
        } catch (NotFoundException $e) {
            // Nothing to do
        }
```

to reduce complexity, right?

*Side: I'm not sure that allowing to manipulate this data by patching some loopholes is the best approach, but I don't have a better one, so for now I'd approve that.*

Reply:
@alongosz what if `setSectionForSubtree` throws `NotFoundException`? I kinda did it intentionally to move this method out of `try-catch`.

Reply:
> what if `setSectionForSubtree` throws `NotFoundException`? I kinda did it intentionally to move this method out of `try-catch`.

Wasn't the point of this entire PR to "patch" possible erroneous conditions?

Reply:
This patch was meant to fix just one of those conditions: the fact that some Locations doesn't have Content objects which breaks some API methods and that is handled by `try-catch` with just the `loadContentInfo` call.

Potentially when I move `setSectionForSubtree` into the `try` it may catch the exception coming out of `setSectionForSubtree` whereas there was no exception for `loadContentInfo` call. So, after all, it will occur that Content is properly loaded (case when Location has Content) but the section is not updated properly - the method will still be finished successfully but it should not at all.

Reply:
Hmm, ok.